component.go

```diff
@@ -71,3 +71,34 @@ func (s *ComponentService) DeleteComponent(ctx context.Context, pageID string, c
 	_, err = s.client.do(ctx, req, nil)
 	return err
 }
+
+// UpdateComponentParams are the parameters that can be changed using the update component API endpoint
+type UpdateComponentParams struct {
+	Description        string    `json:"description,omitempty"`
+	Status             string    `json:"status,omitempty"`
+	Name               string    `json:"name,omitempty"`
+	OnlyShowIfDegraded string    `json:"only_show_if_degraded,omitempty"`
+	GroupID            string    `json:"group_id,omitempty"`
+	Showcase           string    `json:"showcase,omitempty"`
```

Comment:
Looking at the [API docs](https://developer.statuspage.io/#operation/patchPagesPageIdComponentsComponentId) this should be a boolean, no?

```suggestion
	Showcase           bool      `json:"showcase,omitempty"`
```

Reply:
refactored